sonar-batch/src/main/java/org/sonar/batch/bootstrapper/LoggingConfigurator.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.batch.bootstrapper;
+
+import org.sonar.home.log.LogListener;
+
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.classic.Level;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.File;
+
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import org.slf4j.LoggerFactory;
+import org.sonar.core.config.Logback;
+
+public class LoggingConfigurator {
+  private LoggingConfigurator() {
+  }
+
+  public static void apply(LoggingConfiguration conf, File logbackFile) {
+    Logback.configure(logbackFile, conf.substitutionVariables);
+
+    if (conf.listener != null) {
+      setCustomRootAppender(conf);
+    }
+  }
+
+  public static void apply(LoggingConfiguration conf) {
+    apply(conf, "/org/sonar/batch/bootstrapper/logback.xml");
+  }
+
+  public static void apply(LoggingConfiguration conf, String classloaderPath) {
+    Logback.configure(classloaderPath, conf.substitutionVariables);
+
+    // if not set, keep default behavior (configured to stdout through the file in classpath)
+    if (conf.listener != null) {
+      setCustomRootAppender(conf);
+    }
+  }
+
+  private static void setCustomRootAppender(LoggingConfiguration conf) {
+    Logger logger = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1312)